packages/backend/src/core/discovery/saveDiscoveryResult.ts

```diff
@@ -16,21 +29,42 @@ export async function saveDiscoveryResult(
     Object.entries(abis).sort(([a], [b]) => a.localeCompare(b)),
   )
 
-  const project: ProjectParameters = {
+  return {
     name,
     blockNumber,
-    contracts: results
-      .filter((x) => !x.meta.isEOA)
-      .map((x) => ({ ...x, meta: undefined })),
+    contracts: results.filter((x) => !x.meta.isEOA).map(toContractParameters),
     eoas: results
       .filter((x) => x.meta.isEOA)
       .map((x) => x.address)
       .sort((a, b) => a.localeCompare(b.toString())),
     abis,
   }
+}
 
-  await writeFile(
-    `discovery/${name}/discovered.json`,
-    JSON.stringify(project, null, 2),
-  )
+export function toContractParameters(
+  analyzedData: AnalyzedData,
+): ContractParameters {
+  const contract: ContractParameters = {
+    name: analyzedData.name,
+    address: analyzedData.address,
+    upgradeability: analyzedData.upgradeability,
+  }
+
+  if (analyzedData.unverified !== undefined) {
+    contract.unverified = analyzedData.unverified
+  }
+
+  if (analyzedData.code !== undefined) {
+    contract.code = analyzedData.code
```

Comment:
FYI: this changes the slightly the output of already discovered files (order of properties in json)